### PR TITLE
Create a code action class per error code.

### PR DESCRIPTION
**Pull request by angelozerr**
Create a code action class per error code.

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR clean Qute code actions:

 * there is a code action class per error code to avoid having a big class file QuteCodeActions which is hard to read. Splitting code action per error code will simplify the contribution to code action because it will not impact the single QuteCodeAction class.
 * CodeActionRequest wraps all input parameters (template, diagnostic). We ca add another parameters in the future without breaking API. CodeActionRequest provides :
   * getCoveredNode which is the covered node higlighted by the diagnostic range.  
   *  getJavaTypeOfCoveredNode which is the Java type of the covered part.
   * @datho7561 you could add in this class  `canSupportResolve`  method likehttps://github.com/eclipse/lemminx/blob/18e3bdf48a26e0cae6d00fb3f5edb6a4fe1b6708/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/codeaction/ICodeActionRequest.java#L49 when you will support resolve of code action.
 * I simplify data which stores only the Java base type signature (not the source, not the property name) because those informations can be retrieved easily with the convered node (ex : property name) and the Java type of the covered part (ex : source).

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/diagnostics/QuteDiagnosticContants.java [hunk 1]
```diff
@@ -11,17 +11,17 @@
 *******************************************************************************/
 package com.redhat.qute.services.diagnostics;
 
+/**
+ * Constants for Qute diagnostics.
```
Comment (datho7561) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/diagnostics/QuteDiagnosticContants.java:15]: I think you misspelled your name

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/AbstractQuteCodeAction.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+	/**
+	 * Generate code action(s) sor similar values (Did you mean ...?)
+	 * 
+	 * @param part            the part node.
+	 * @param availableValues the available values.
+	 * @param template        the Qute template.
+	 * @param diagnostic      the diagnostic.
+	 * @param codeActions     list of CodeActions.s
+	 */
```
Comment (datho7561) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/AbstractQuteCodeAction.java:64]: Small typos:
* sor -> for
* The "s" after the "." for the param codeActions

**Review by datho7561 — CHANGES_REQUESTED** [review]
Looks really good. I like the code structure, and I think this will help when we need to implement more code actions. I caught a few small documentations things, and once they are addressed, I think it should be good to merge.

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/AbstractQuteCodeAction.java (hunk 2 above)
Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/AbstractQuteCodeAction.java]: fixed

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/diagnostics/QuteDiagnosticContants.java (hunk 1 above)
Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/diagnostics/QuteDiagnosticContants.java:15]: fixed

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/AbstractQuteCodeAction.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+				codeActions.add(similarCodeAction);
+			}
+		}
+	}
+
+	/**
+	 * CodeAction to change severity setting value to "ignore"
+	 *
+	 * @param template        the Qute template.
+	 * @param diagnostics     a singleton list diagnostic to set to ignore.
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/codeactions/AbstractQuteCodeAction.java]: fixed

**Review by AlexXuChen — COMMENTED** [review]
LGTM, can confirm all CodeActions are still working. One thing I noticed is that there is a flicker for "Quick Fix..." pnly when hovering any`UnknownProperty` diagnostic:
Peek 2022-08-02 09-58
However, all `UnknownProperty` CodeActions are available and working, seems to be a small performance issue when generating the code action.

File: qute.jdt/com.redhat.qute.jdt.test/projects/maven/qute-quickstart/src/main/resources/templates/hello.qute.html [hunk 4]
```diff
@@ -2,18 +2,9 @@
 <html>
 <head>
 <meta charset="UTF-8">
-<title>Qute Hello World</title>
+<title>Qute Hello World</title> 
 </head>
 <body>
     <p>Hello {name ?: "world"}!</p>
 </body>
-</html>
```
Comment (datho7561) [inline qute.jdt/com.redhat.qute.jdt.test/projects/maven/qute-quickstart/src/main/resources/templates/hello.qute.html:10]: Any reason this is being changed?

Comment (AlexXuChen) [reply qute.jdt/com.redhat.qute.jdt.test/projects/maven/qute-quickstart/src/main/resources/templates/hello.qute.html:10]: I accidentally pushed a change to this test file in my CodeAction pr, I just asked Angelo if he could revert it.

**Review by datho7561 — APPROVED** [review]

**Pull request merged**